src/privatesend-server.cpp

```diff
@@ -174,12 +174,14 @@ void CPrivateSendServer::ProcessMessage(CNode* pfrom, const std::string& strComm
         if (entry.vecTxDSIn.size() > PRIVATESEND_ENTRY_MAX_SIZE) {
             LogPrintf("DSVIN -- ERROR: too many inputs! %d/%d\n", entry.vecTxDSIn.size(), PRIVATESEND_ENTRY_MAX_SIZE);
             PushStatus(pfrom, STATUS_REJECTED, ERR_MAXIMUM, connman);
+            ConsumeCollateral(connman, entry.txCollateral);
             return;
         }
 
         if (entry.vecTxOut.size() > PRIVATESEND_ENTRY_MAX_SIZE) {
             LogPrintf("DSVIN -- ERROR: too many outputs! %d/%d\n", entry.vecTxOut.size(), PRIVATESEND_ENTRY_MAX_SIZE);
             PushStatus(pfrom, STATUS_REJECTED, ERR_MAXIMUM, connman);
+            ConsumeCollateral(connman, entry.txCollateral);
             return;
         }
```

Comment:
Do we never check if vecTxDSIn == vecTxDSOut?

Reply:
Good point (assuming you were talking about their size) - fc853b176

Reply:
based on that, doing this check is kindof redundant, if they are equal there's no need to check both

Reply:
Re: max size check? There's still a max size limit. You could have equal numbers of inputs / outputs, but have too many of them (e.g. 50 inputs / 50 outputs).